### ISSUE TITLE
Fix NPM audit advisory 745

### ISSUE THREAD

**Issue by philmander** [Ziggeo/ziggeo-client-sdk#21]
Could you please fix this NPM vulnerability caused by `underscore.string`: https://www.npmjs.com/advisories/745

Output from NPM Audit
```
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ underscore.string                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.3.5                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ ziggeo-client-sdk                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ ziggeo-client-sdk > grunt-yaml > js-yaml > argparse >        │
│               │ underscore.string                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/745                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by Bane-D** [comment]
Hi Phil,

Thank you for reporting this. We are looking into it to see how it can be upgraded or exchanged with a different package so that it passes the check.

Having said that I do want to mention that this should only be used during the build. Specifically the grunt-yaml is something that would be used during the build and would not be included within the live/built version.

Of course as mentioned, while it is not included we still prefer all green lights and are looking into it.

Regards,
Bane

**Comment by philmander** [comment]
I don't know much about your build process, but `grunt-yaml` is listed in your `package.json`'s regular `dependencies` and not `devDependencies`. 

Maybe this is why, as a consumer, it reports this to me.

**Comment by ntshcalleia** [comment]
Fixed on R34 (d0de407832fc869199cce8227ec67cbb2f4bc250). Run `npm audit --production` to ignore dev dependencies.
